src/main/java/org/opensearch/ad/util/ParseUtils.java

```diff
@@ -562,13 +562,13 @@ public static void onGetAdResponse(
                     function.accept(detector);
                 } else {
                     logger.debug("User: " + requestUser.getName() + " does not have permissions to access detector: " + detectorId);
-                    listener.onFailure(new OpenSearchException("User does not have permissions to access detector: " + detectorId));
+                    listener.onFailure(new AnomalyDetectionException("User does not have permissions to access detector: " + detectorId));
                 }
             } catch (Exception e) {
-                listener.onFailure(new OpenSearchException("Unable to get user information from detector " + detectorId));
+                listener.onFailure(new AnomalyDetectionException("Unable to get user information from detector " + detectorId));
             }
         } else {
-            listener.onFailure(new ResourceNotFoundException("AnomalyDetector is not found with id: " + detectorId));
+            listener.onFailure(new ResourceNotFoundException(detectorId, "AnomalyDetector is not found with id: " + detectorId));
```

Comment:
Can we use `CommonErrorMessages.FAIL_TO_FIND_DETECTOR_MSG` here? And is it worth refactoring some of the other messages into constants so they can be reused?

Reply:
I'm refactoring to put the general message into constants.

Reply:
done